hydromt/drivers/pyogrio_driver.py

```diff
@@ -0,0 +1,90 @@
+"""Driver to read geodataframes using Pyogrio."""
+from logging import Logger
+
+import geopandas as gpd
+from pyogrio import read_dataframe, read_info
+from pyproj import CRS
+from shapely.geometry.base import BaseGeometry
+
+from hydromt.gis_utils import parse_geom_bbox_buffer
+from hydromt.typing import GEOM_TYPES, GPD_TYPES
+
+# from hydromt.nodata import NoDataStrategy
+from .geodataframe_driver import GeoDataFrameDriver
+
+
+class PyogrioDriver(GeoDataFrameDriver):
+    """Driver to read GeoDataFrames using the `pyogrio` package."""
+
+    def read(
+        self,
+        uri: str,
+        bbox: list[int] | None = None,
```

Comment:
Bbox definitely has a type in typing.py. If we're using validated pydantic models this is fine, if not I think we ougt to prefer tuples because for those we can assert their length in the type itself .